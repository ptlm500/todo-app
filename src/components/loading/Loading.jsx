import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import { theme } from '../../styles';

export const load = () => keyframes`
    0% {
      opacity: 0.5;
      height: 20%;
    }
    50% {
      opacity: 1;
      height: 100%;
    }
    100% {
      opacity: 0.5;
      height: 20%;
    }
`;

export default function Loading({size}) {
  return (
    <LoadingWrapper size={size}>
      <div />
      <div />
      <div />
      <div />
    </LoadingWrapper>
  );
}

Loading.propTypes = {
  size: PropTypes.number
};

const defaultSize = 120;

const LoadingWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: ${props => props.size || defaultSize}px;
  height: ${props => props.size || defaultSize}px;

  > div {
    background: ${theme.colour.primary};
    width: 20%;
    height: 20%;
    border-radius: ${theme.defaultRadius};
    animation: ${load()} 2s ease-in-out infinite;
    opacity: 0.1;
  }

  > div:nth-child(1) {
    animation-delay: -1.5s;
  }

  > div:nth-child(2) {
    animation-delay: -1s;
  }

  > div:nth-child(3) {
    animation-delay: -0.5s;
  }

  > div:nth-child(4) {
    animation-delay: 0s;
  }
`;
