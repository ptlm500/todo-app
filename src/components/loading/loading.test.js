import React from 'react';
import renderer from 'react-test-renderer';
import Loading, { load } from './Loading';
import 'jest-styled-components';

describe('Loading', () => {
  it('matches the snapshot', () => {
    expect(renderer.create(<Loading />)).toMatchSnapshot();
  });

  it('matches the snapshot when a custom size is provided', () => {
    expect(renderer.create(<Loading size={10} />)).toMatchSnapshot();
  });
});

describe('load', () => {
  it('matches the snapshot', () => {
    expect(load()).toMatchSnapshot();
  });
});
