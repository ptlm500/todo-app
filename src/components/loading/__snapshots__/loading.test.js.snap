// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading matches the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 120px;
  height: 120px;
}

.c0 > div {
  background: #3F51B5;
  width: 20%;
  height: 20%;
  border-radius: 0.5rem;
  -webkit-animation: jltStV 2s ease-in-out infinite;
  animation: jltStV 2s ease-in-out infinite;
  opacity: 0.1;
}

.c0 > div:nth-child(1) {
  -webkit-animation-delay: -1.5s;
  animation-delay: -1.5s;
}

.c0 > div:nth-child(2) {
  -webkit-animation-delay: -1s;
  animation-delay: -1s;
}

.c0 > div:nth-child(3) {
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

.c0 > div:nth-child(4) {
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

<div
  className="c0"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Loading matches the snapshot when a custom size is provided 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 10px;
  height: 10px;
}

.c0 > div {
  background: #3F51B5;
  width: 20%;
  height: 20%;
  border-radius: 0.5rem;
  -webkit-animation: jltStV 2s ease-in-out infinite;
  animation: jltStV 2s ease-in-out infinite;
  opacity: 0.1;
}

.c0 > div:nth-child(1) {
  -webkit-animation-delay: -1.5s;
  animation-delay: -1.5s;
}

.c0 > div:nth-child(2) {
  -webkit-animation-delay: -1s;
  animation-delay: -1s;
}

.c0 > div:nth-child(3) {
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

.c0 > div:nth-child(4) {
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

<div
  className="c0"
  size={10}
>
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`load matches the snapshot 1`] = `
Keyframes {
  "id": "sc-keyframes-jltStV",
  "inject": [Function],
  "name": "jltStV",
  "stringifyArgs": Array [
    "
    0% {
      opacity: 0.5;
      height: 20%;
    }
    50% {
      opacity: 1;
      height: 100%;
    }
    100% {
      opacity: 0.5;
      height: 20%;
    }
",
    "jltStV",
    "@keyframes",
  ],
  "toString": [Function],
}
`;
